src/meshapi/admin/ranked_search.py

```diff
@@ -0,0 +1,41 @@
+import typing
+from typing import Optional, Tuple
+
+from django.contrib.admin import ModelAdmin
+from django.contrib.admin.views.main import ORDER_VAR
+from django.contrib.postgres.search import SearchQuery, SearchRank, SearchVector
+from django.db.models import QuerySet
+from django.db.models.expressions import CombinedExpression
+from django.http import HttpRequest
+
+# Trick stolen from https://stackoverflow.com/a/56991089 to make mypy happy about mixin types
```

Comment:
I wonder if I could use that on the button thing I was having trouble with the other day.

Reply:
I don't think so. I thought about this then. The reason this is needed is that for MixIns, the parent class is unknown to the mypy at declaration time, so we use this trick to tell it what the parent class will eventually be. In your situation, mypy knew the parent class, but was wrong about the availability of its methods